spec/lib/rbac/filterer_spec.rb

```diff
@@ -1286,56 +1287,143 @@ def get_rbac_results_for_and_expect_objects(klass, expected_objects)
         end
       end
 
-      it "returns users from current user's groups" do
-        other_user.miq_groups << group
-        get_rbac_results_for_and_expect_objects(User, [user, other_user])
-      end
+      context "with other users in the group" do
+        before do
+          other_user.miq_groups << group
+        end
+
+        it "returns users from current user's groups" do
+          get_rbac_results_for_and_expect_objects(User, [user, other_user])
+        end
 
-      it "returns user's groups" do
-        _expected_groups = [group, other_group] # this will create more groups than 2
-        get_rbac_results_for_and_expect_objects(MiqGroup, user.miq_groups)
+        it "returns user's groups" do
+          get_rbac_results_for_and_expect_objects(MiqGroup, [group])
+        end
       end
 
-      context "with self-service user" do
+      context "where the user is in another group that is a regular group" do
```

Comment:
I really like using when for contexts

```suggestion
      context "when the user is in another group that is a regular group" do
```

Reply:
Yeah I was debating the wording - I like the latter there.